### DISCUSSION TITLE
Serial Monitor Windows Arduino IDE Adafruit QT Py RP2040

### DISCUSSION THREAD

**Discussion by KerseyFabrications** [earlephilhower/arduino-pico#154] [Q&A] [answered]
I'm unable to get the Arduino IDE serial monitor to work under Windows using the Adafruit QT Py RP2040. Sketches compile and install fine. I see COM5 for the device but if I try to open COM5, I just get "Port Busy." Putty is unable to open it either. Is this a Windows issue or something else?

Thanks!

**Comment by earlephilhower** [comment, marked as answer]
Windows does run fine for quite a few of the users that have reported back, so it could just be a configuration issue.

Are you running Win7 or Win10?  With Win7, the Zadig utility must be used (see the readthedocs site for images) and configured to get the right driver for the Pico installed, or the serial port won't be accessible.  I don't even have any remaining W7 VMs so can't 

For Win 10, there really haven't been any problems noted.  It just works fine on my laptops without any special drivers or configurations.

Maybe some other users have found/fixed something?

**Reply by KerseyFabrications** [reply]
Sorry to bother you. I figured it out. 3D printing programs (Cura) can be greedy and they grab the serial port without permission. I usually catch this but forgot about it this time. Thank you for the work on this package.

**Reply by Gavin-Perry** [reply]
Thanks Earl for a great board manager for RP2040!
I think this is similar to my issue. I started a new thread but maybe more appropriate here.
Generic RP2040 Zero (from WaveShare)  Windows 11 (Aspire 5) Arduino IDE 1.8.16 Arduino-Pico
Programs load and run but give error of not finding COMx if it is still open from the last run.
If I close and reopen Serial Monitor it starts working. Most (AVR, ESP Teensie ... ) work fine reattaching to an open COM

**Reply by earlephilhower** [reply]
The thing w/the Pico is the COM port is USB and controlled by the Pico itself.  So it goes away when you upload (replaced w/the UF2 upload drive) and then appears again once the Pico reboots after program upload and starts emulating a COM port again.

I believe those other devices have a dedicated USB Serial chip which is always there and connected, even when the main microprocessor gets rebooted.

**Comment by Gavin-Perry** [comment]
Thank for the explanation. I guess I can manage closing the COM when I
compile and upload.
Would it be possible for the IDE to close and open the COM when an upload
is started? (Something to ask of arduino.cc staff?)

While I have you another problem I have is finding .h files here is the
error message
In file included from
C:\Users\gavin\AppData\Local\Temp\.arduinoIDE-unsaved2022119-16704-nvnk5r.zf8b9\RGB_input\RGB_input.ino:11:
c:\Users\gavin\OneDrive\Documents\Arduino\libraries\Pico-DMX\src/DmxInput.h:13:10:
fatal error: pio.h: No such file or directory
   13 | #include <pio.h>
      |          ^~~~~~~
compilation terminated.
Compilation error: exit status 1}
If I select Raspberry Pi Pico as my board rather than Generic Pico program
compiles OK
I just upgraded to 2.0 RC3 and that didn't help though now I get choice of
PicoProbe or pico-debug for the board.
Suggestions of which to choose if I don't really need a fancy debugger?

Thanks for your support. Do you have a buymeacoffee.com page?
Easy way to get feedback and some support for your efforts :-)
Here's me
https://www.buymeacoffee.com/GavinPerry

Thanks again for the Pico package.


--
Gavin Perry, PhD
Chief Technology Officer
Meridian Electric Company
2392 Grissom Drive
St Louis MO 63146
http://meridianlighting.com
cell: 314 406-0697 <http://voice.google.com/calls?a=nc,%2B13144060697>

**Reply by earlephilhower** [reply]
You need to include the path as well:
````
#include <hardware/pio.h>
````

As for the Serial Monitor, that's all in the IDE and nothing the core can do AFAIK.

**Comment by Gavin-Perry** [comment]
As an extra include where?


Gavin Perry

**Reply by Andy2No** [reply]
> As an extra include where? Gavin Perry

Instead of the include it complains about, on line 13:

   13 | #include <pio.h>

Change that to; 

#include <hardware/pio.h>

**Comment by khoih-prog** [comment]
Hi @Gavin-Perry 

I just created the PR [**Add support to arduino-pico and mbed_nano core** #24](https://github.com/jostlowe/Pico-DMX/pull/24) to [Pico-DMX library](https://github.com/jostlowe/Pico-DMX) to

1. Add support to [Earle Philhower's arduino-pico core](https://github.com/earlephilhower/arduino-pico)
2. Add support to `Nano_RP2040_Connect` using `mbed_nano` core

---

Compile Results for `Nano_RP2040_Connect`using [Earle Philhower's arduino-pico core](https://github.com/earlephilhower/arduino-pico)

![Selection_104](https://user-images.githubusercontent.com/57012152/154868929-2a33114c-b3c2-4e86-925b-c726e680d0f4.png)

**Comment by Gavin-Perry** [comment]
You guys are great!
I'll try it all out tomorrow and let you know how it's working.
I am so glad they are letting me use open source on this project. I would
never get this good of support from commercial software companies. Fixed
over the weekend!  :-)

Gavin Perry
